include/communicator_quda.h

```diff
@@ -0,0 +1,803 @@
+#pragma once
```

Comment:
Making a comment for myself, if nothing else.  Moving all these functions into the header is a bit problematic since this places a lot of CUDA dependencies into headers - a problem for portable QUDA.  The `isHost` and `comm_peer2peer_init` This is something that can probably be taken care of when this is merged into the generic_kernels branch.